app/scripts/controllers/preferences.js

```diff
@@ -226,6 +227,15 @@ export default class PreferencesController {
     this.store.updateState({ useCurrencyRateCheck: val });
   }
 
+  /**
+   * Setter for the `useRequestQueue` property
+   *
+   * @param {boolean} val - Whether or not the user wants to have requests queued if network change is required.
+   */
+  setUseRequestQueue(val) {
+    this.store.updateState({ useRequestQueue: val });
```

Comment:
I think we need to listen for updates to this, and on change, we should re-call `setupProviderEngine`. If we dont, toggleing the request queue option will leaave the wallet in a state where its using the 'non-domain-specific provider'

Reply:
Confirmed that this is a bit of a bug. turning the feature flag on / off  requires  refreshing of the  dapp for it to take effect.

Reply:
is this something we want to address now or later?

Reply:
I think this is ok while behind the experimental toggle since its an edge case that would happen max once when users turn it on. But we should resolve it before we ship it fully.